### PR TITLE
Windows port

### PR DESCRIPTION
**Pull request by dra27**
There isn't an issue for Windows support at the moment, so here's a pull request to track it in instead. It's obviously not yet intended for merging.

There's a squashed commit of changes from #172 and this backports https://github.com/ocaml/ocaml/pull/1721.

All comments and contributions welcome!

### CODE REVIEW

File: byterun/win32.c [hunk 1]
```diff
@@ -914,3 +916,22 @@ void caml_restore_win32_terminal(void)
...
+  QueryPerformanceFrequency(&frequency);
+}
+
+int64_t caml_time_counter(void)
+{
+  LARGE_INTEGER now;
+  /* Windows 2000 is no longer supported, so this function always succeeds */
+  QueryPerformanceCounter(&now);
+  now.QuadPart *= 1000000000L;
+  return (now.QuadPart * 1000000000L) / frequency.QuadPart;
```
Comment (stedolan) [inline byterun/win32.c]: This doesn't look right. First, it does the sec->nanosecs conversion twice. Second, even if only done once, I think the multiplication will overflow after less than a day.

**Comment by kayceesrk** [comment]
PR seems to have broken native compilation. See travis logs: https://travis-ci.org/ocamllabs/ocaml-multicore/jobs/365689997

> Minor_heap_align_bits misconfigured for this platform

File: byterun/win32.c (hunk 1 above)
Comment (dra27) [reply byterun/win32.c]: Yeah, I seem to have turned my brain off there.

File: byterun/win32.c [hunk 2]
```diff
@@ -925,13 +925,14 @@ void caml_init_os_params(void)
   GetSystemInfo(&si);
   caml_sys_pagesize = si.dwPageSize;
   QueryPerformanceFrequency(&frequency);
+  /* Convert the frequency to the duration of 1 tick in ns */
+  frequency.QuadPart = 1000000000L / frequency.QuadPart;
```
Comment (dra27) [inline byterun/win32.c:929]: @stedolan 128bit arithmetic will be bad - is converting the tick frequency to ns acceptable?

**Pull request closed without merging**